cid/helpers/quicksight/__init__.py

```diff
@@ -341,7 +352,7 @@ def list_data_sources(self) -> list:
                 return result.get('DataSources')
         except self.client.exceptions.AccessDeniedException:
             logger.info('Access denied listing data sources')
-            return list()
+            raise
         except Exception as e:
             logger.debug(e, stack_info=True)
             return list()
```

Comment:
Would it be more logical to raise here as well?

Reply:
Maybe, but will requires to change in other places as well